outputs/jira_test.go

```diff
@@ -0,0 +1,1026 @@
+			wantError: "Jira Cloud can't work with PAT",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			client, err := test.jiraApi.buildTransportClient()
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				assert.Equal(t, reflect.TypeOf(test.wantTransport), reflect.TypeOf(client.Transport))
+			}
+		})
+	}
+}
+
+func TestJiraApi_createClient(t *testing.T) {
+	tests := []struct {
+		name      string
+		jiraApi   *JiraAPI
+		wantError string
+	}{
+		{
+			name:    "happy path",
+			jiraApi: &JiraAPI{},
+		},
+		{
+			name:      "sad path (using PAT for cloud jira)",
+			jiraApi:   &JiraAPI{Token: "token", Url: "https://johndoe.atlassian.net"},
+			wantError: "Jira Cloud can't work with PAT",
+		},
+		{
+			name:      "sad path (bad url)",
+			jiraApi:   &JiraAPI{Url: "https://johndoe   .atlassian.net"},
+			wantError: "unable to create new JIRA client",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := createClient(test.jiraApi)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestJiraAPI_Send(t *testing.T) {
+	tests := []struct {
+		name           string
+		jiraApi        *JiraAPI
+		createMetaInfo *jira.CreateMetaInfo
+		fieldList      *[]jira.Field
+		priorityList   *[]jira.Priority
+		issue          *jira.Issue
+		content        map[string]string
+		wantError      string
+	}{
+		{
+			name: "happy path",
+			jiraApi: &JiraAPI{
+				ProjectKey:      "project",
+				User:            "user",
+				boardType:       "scrum",
+				Labels:          []string{"label1", "label2"},
+				FixVersions:     []string{"fix1", "fix2"},
+				AffectsVersions: []string{"affect1", "affect2"},
+			},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			fieldList:      fieldList,
+			priorityList:   &[]jira.Priority{{Name: "High"}},
+			issue:          &jira.Issue{},
+			content:        map[string]string{"title": "title_content", "description": "description_content"},
+		},
+		{
+			name:      "sad path (Failed to create client)",
+			wantError: "Failed to create client",
+		},
+		{
+			name:      "sad path (Failed to create meta project)",
+			jiraApi:   &JiraAPI{},
+			wantError: "Failed to create meta project",
+		},
+		{
+			name:           "sad path (Failed to get issuetype)",
+			jiraApi:        &JiraAPI{Issuetype: "Bug", ProjectKey: "project"},
+			createMetaInfo: &jira.CreateMetaInfo{Projects: []*jira.MetaProject{{Key: "project", IssueTypes: []*jira.MetaIssueType{metaIssuetype}}}},
+			wantError:      "Failed to get issuetype",
+		},
+		{
+			name:           "sad path (Failed to create fields config)",
+			jiraApi:        &JiraAPI{ProjectKey: "project"},
```

Comment:
ditto but for missing project key?

Reply:
We test for an empty ProjectKey in [this test](https://github.com/aquasecurity/postee/blob/4e0e0efbeddb93ee56285587da85a74182ff91c6/outputs/jira_test.go#L511-L515).